src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -61,6 +65,61 @@ public IAnyCallConfigurationWithNoReturnTypeSpecified CallTo(object fakeObject)
             return this.configurationFactory.CreateAnyCallConfiguration(manager, rule);
         }
 
+        public IVoidArgumentValidationConfiguration CallToSet<TValue>(Expression<Func<TValue>> propertySpecification)
+        {
+            Guard.AgainstNull(propertySpecification, nameof(propertySpecification));
+            this.AssertThatMemberCanBeIntercepted(propertySpecification);
+
+            var setterExpression = BuildSetterFromGetter(propertySpecification);
+            var fake = this.expressionParser.GetFakeManagerCallIsMadeOn(setterExpression);
+            var rule = this.ruleFactory.Invoke(setterExpression);
+
+            fake.AddRuleFirst(rule);
+
+            return this.configurationFactory.CreateConfiguration(fake, rule);
+        }
+
+        private string GetExpressionDescription<TValue>(Expression<Func<TValue>> expression)
+        {
+            var matcher = new ExpressionCallMatcher(
+                expression,
+                ServiceLocator.Current.Resolve<ExpressionArgumentConstraintFactory>(),
+                ServiceLocator.Current.Resolve<MethodInfoManager>(),
+                this.callExpressionParser);
+
+            var expressionDescription = matcher.DescriptionOfMatchingCall;
+            return expressionDescription;
+        }
+
+        private LambdaExpression BuildSetterFromGetter<TValue>(Expression<Func<TValue>> propertySpecification)
+        {
+            MemberExpression memberExpression = propertySpecification.Body as MemberExpression;
+            if (memberExpression == null)

```

Comment:
That won't work for indexed properties, because in that the body is a call to `get_TheProperty(index)` (of course in C# `TheProperty` is always `Item`, since named indexers are not allowed, but in VB it can be anything)